Week3/homework/step2-6.js

```diff
@@ -4,13 +4,25 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArray1 = [];
+  arr.forEach(element => {
+    element.forEach(item => {
+      newArray1.push(item);
+    });
+  });
+  return newArray1;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArray2 = [];
+  arr.forEach(element => {
+    element.forEach(child =>
+      child.forEach(item => {
+        newArray2.push(item);
```

Comment:
Nice work! Though I'm usually a little wary when I see more than 2 levels of nested loops - the fewer loops the better. If you'd like you could re-think this using recursion?

Reply:
Yes, I will keep on seperate, not nested